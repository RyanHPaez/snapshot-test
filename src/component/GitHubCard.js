import React, { useEffect, useState } from "react";
import Card from "react-bootstrap/Card";

function GitHubCard() {
  const [gitHubName, setGitHubName] = useState("");

  return (
    <div className="GitHubCard">
      <h1>Ryan's Github Profile Info:</h1>
      <h2>{gitHubName}</h2>
      <Card
        style={{
          width: "100%",
          backgroundImage:
            "url(https://avatars.githubusercontent.com/u/93347623?v=4)",
          height: "500px",
          backgroundRepeat: "no-repeat",
          backgroundPosition: "center",
          backgroundColor: "lightblue",
          backgroundSize: "200px 200px",
        }}
      >
        <Card.Subtitle style={{ fontSize: "15px" }} className="paragraph">
          {" "}
          Self-belief and hard work will always earn you success.{" "}
        </Card.Subtitle>
      </Card>
    </div>
  );
}

export default GitHubCard;
