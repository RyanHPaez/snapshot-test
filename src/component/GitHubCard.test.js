
import React from "react";
import GitHubCard from "./GitHubCard";




test("receives GitHub name from GitHub REST API using jest fetch mock", async () => {
  const tree = render(<GitHubCard/>)
  expect(tree).toMatchSnapshot()
    const gitHubName = await waitFor(() => screen.getByRole('heading', { level: 2 }))
    expect(gitHubName)
  })

