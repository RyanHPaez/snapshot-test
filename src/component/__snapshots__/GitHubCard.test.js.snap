// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`receives GitHub name from GitHub REST API using jest fetch mock 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="GitHubCard"
      >
        <h1>
          Ryan's Github Profile Info:
        </h1>
        <h2 />
        <div
          class="card"
          style="width: 100%; background-image: url(https://avatars.githubusercontent.com/u/93347623?v=4); height: 500px; background-repeat: no-repeat; background-position: center; background-color: lightblue; background-size: 200px 200px;"
        >
          <div
            class="paragraph card-subtitle h6"
            style="font-size: 15px;"
          >
             
            Self-belief and hard work will always earn you success.
             
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="GitHubCard"
    >
      <h1>
        Ryan's Github Profile Info:
      </h1>
      <h2 />
      <div
        class="card"
        style="width: 100%; background-image: url(https://avatars.githubusercontent.com/u/93347623?v=4); height: 500px; background-repeat: no-repeat; background-position: center; background-color: lightblue; background-size: 200px 200px;"
      >
        <div
          class="paragraph card-subtitle h6"
          style="font-size: 15px;"
        >
           
          Self-belief and hard work will always earn you success.
           
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
